eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -98,6 +98,10 @@ object Channel {
    */
   case class OutgoingMessage(msg: LightningMessage, peerConnection: ActorRef)
 
+  def replyToCommand(sender: ActorRef, reply: CommandResponse[Command], cmd: Command): Unit = cmd match {
+    case cmd1: HasReplyToCommand => if (cmd1.replyTo == ActorRef.noSender) sender ! reply else cmd1.replyTo ! reply
+    case cmd1: HasOptionalReplyToCommand => cmd1.replyTo_opt.foreach(_ ! reply)
+  }
```

Comment:
I'm not sure this makes sense in the `object`, we've put everything in the `class` up until now.
Let's see what @pm47 thinks and why he didn't add such a method initially.

Reply:
> Let's see what @pm47 thinks and why he didn't add such a method initially.

My initial goal was to make all commands a `HasReplyTo`

> I'm not sure this makes sense in the object, we've put everything in the class up until now.

There is a major difference : the `!` method has an implicit argument that would be lost if the method is defined in the `object`. We may want to do it on purpose though, since with _akka typed_ there is no more `sender`.
```scala
def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit
```

Reply:
so... can we proceed with this then?

Reply:
I don't think so, this is a change in behavior since the implicit `sender` is lost when putting the function in the `object`.
I would not change this for now, and wait for our full migration to akka typed which will completely remove that code.